javamelody-core/src/main/java/net/bull/javamelody/internal/model/CollectorServer.java

```diff
@@ -105,33 +100,50 @@ public void run() {
 	public void collectWithoutErrors() {
 		// clone pour éviter ConcurrentModificationException
 		final Map<String, List<URL>> clone;
+		final Map<String, String> urlHeaders;
 		try {
 			clone = new LinkedHashMap<String, List<URL>>(
 					Parameters.getCollectorUrlsByApplications());
+			urlHeaders = Parameters.getCollectorHeadersByApplications();

```

Comment:
Clone is need for headers like for URLs : `new LinkedHashMap<String, List<URL>>(headers)`
And by definition, URL has no headers. Http request has headers.